packages/flutter/lib/src/widgets/scroll_position.dart

```diff
@@ -895,6 +895,12 @@ abstract class ScrollPosition extends ViewportOffset with ScrollMetrics {
     UserScrollNotification(metrics: copyWith(), context: context.notificationContext!, direction: direction).dispatch(context.notificationContext);
   }
 
+  /// Dispatches a notification that the [ScrollMetrics] has changed.
+  @visibleForTesting
+  void sendMetrics(ScrollMetrics metrics) {
+    activity!.dispatchScrollStartNotification(metrics, context.notificationContext);
+  }
+
```

Comment:
I don't know that this is a good idea to add to ScrollPosition, but it has given me an idea that may work. Sometimes we'll create test classes just to facilitate methods like these. Instead of adding this here, in your tests you could create a TestScrollPosition class that instead just overrides one of the existing methods that dispatches a scroll notification and always passes null metrics, or whichever case you are trying to test. WDYT?

Reply:
Yes, that's a great idea, I add these classes to the test, thanks.